presto-cli/src/main/java/com/facebook/presto/cli/Console.java

```diff
@@ -92,7 +92,7 @@ public void run()
         boolean hasQuery = !Strings.isNullOrEmpty(clientOptions.execute);
         boolean isFromFile = !Strings.isNullOrEmpty(clientOptions.file);
 
-        if (!hasQuery || !isFromFile) {
+        if (!hasQuery && !isFromFile) {
```

Comment:
This should be based on whether the output is a tty. Looking at jansi's source code, it would appear to be making that check inside AnsiConsole.wrapOutputStream(), so we should see whether that's working correctly.